### PR TITLE
added multi search v2

### PR DESCRIPTION
**Pull request by Erryk20**
@SergeyKondratovec @Viktorminator @comaw @yurikzt @SergiiBondarchuk @jusnak3 @Alrok

### CODE REVIEW

File: src/Lib/Filters/ElasticSearch/MultiFullTextFilter.php [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+    /**
+     * @param array $indexes
+     * @return $this
+     */
+    public function index($index)
+    {
+        /** @var ElasticSearchFilterElement $index */
+        if (is_object($index)) {
+            $this->index = $index->getValues();;
```
Comment (comaw) [inline src/Lib/Filters/ElasticSearch/MultiFullTextFilter.php]: Correct - ;; -> ;

File: src/Resources/ElasticSearch/ElasticSearchResource.php [hunk 2]
```diff
@@ -101,7 +101,23 @@ protected function getQueryParams(array $additionalParams = [])
         unset($params['offset']);
         unset($params['limit']);
 
-        $where = $additionalParams ? $additionalParams : json_decode($params['where'], true);
+        if ($additionalParams) {
+            $where = $additionalParams;
+        } else {
+            if (!empty($this->filter) && $this->multi) {
+                $where = explode("\n", trim($params['where']));
+                foreach ($where AS $key => &$item) {
```
Comment (comaw) [inline src/Resources/ElasticSearch/ElasticSearchResource.php]: Do not useing links - &$item
Use $where[$key] = $item

File: src/Lib/Http/ElasticSearchResponse.php [hunk 3]
```diff
@@ -22,29 +22,43 @@
...
             $this->included = [];
             $this->meta     = null;
         } catch (\Exception $ex) {
+            $this->success = false;
             $this->errors  = $ex->getMessage();
             throw new \Exception($ex->getMessage());
         }
     }
+    
+    public function getMulti(){
```
Comment (SergiiBondarchuk) [inline src/Lib/Http/ElasticSearchResponse.php]: Use correct formatting

File: src/Lib/Makers/ElasticSearchMaker.php [hunk 4]
```diff
@@ -74,7 +74,21 @@ protected function getResultFromResponse(ElasticSearchResponse $response)
...
+                $this->getDataFromResponse($item, $result);
+            }
+        } else {
+            $this->getDataFromResponse($response, $result);
+        }
+
+        return $result;
+    }
+
+    protected function getDataFromResponse(array $response, &$result)
```
Comment (SergiiBondarchuk) [inline src/Lib/Makers/ElasticSearchMaker.php]: Could we rewrite it without passing by reference?

**Review by ghost — APPROVED** [review]

**Pull request merged**